src/aws.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @typedef {object} ParsedTopic
+ * @property {string|undefined} region - aws region of topic
+ */
+
+/**
+ * Parse an AWS SNS TopicArn
+ * @param {string} topicArn - e.g. arn:aws:sns:us-east-2:foo:topicName
+ * @returns {ParsedTopic}
+ */
+function parseTopic(topicArn) {
+  const pattern = /arn:aws:sns:([^:]*)/i
+  const region = topicArn.match(pattern)?.[1]
```

Comment:
Throw if not matched?

Reply:
I intentionally want to support topics that we dont know how to parse a region from (i.e. the result of parsing is `undefined`). e.g. in a lot of the existing code I've seen the `config.js` files default some of these aws arn values to just the string literal `"thingQueue"` (or equiv). Which tbh I think is nice for flexibility e.g. in testing. 

anyway I'd prefer to not throw here. I can jsdoc about what returning undefined indicates.

Reply:
Why do you want to support topics without a region...is there a case beyond tests? Doesn't the SNS client require one?

Reply:
> Why do you want to support topics without a region

I don't feel to strongly about it, but for me its more like I couldn't come up with a great reason not to support those values (e.g. because they could be using the topics/regions with a customized `snsClient` parameter, not necessarily the aws one. i think of [Postel's "be liberal in what you accept"](https://en.wikipedia.org/wiki/Robustness_principle)

> is there a case beyond tests?

not really.

This seems like a style thing, which I don't have strong opinions on. FWIW this was my thinking: this `parseTopic` function results in a parsing of the region. That result is either a `string` or an indicator that a region could not be parsed. That indicator is indicated by it being of type `undefined`. If instead we want it to only ever return a `string`, but instead `throw` to indicate that 'result' of not being able to parse a region, sure, we can do that too. I tend to just use the return type to indicate stuff until things get complicated enough to really need the caller to maybe wrap in a `try` block.

So I'm happy to `throw` in here if thats what you want. Otherwise I think it might be fine not to. If someone wants to `catch`, e.g. they can bring-their-own `assert` and assert the truthiness/stringiness of the return type of their invocation of this `parseTopic` fn.

Reply:
@alanshaw FWIW I plugged in the `@aws-sdk/util-arn-parser` and it made it so this `parseTopic` now throws if it cant parse the topic as an arn